### ISSUE TITLE
mysql server not start after creating container.

### ISSUE THREAD

**Issue by vsjadeja** [bitnami/bitnami-docker-mysql#45]
**Description**
I am running this command for creating mysql container for master database:

```
docker run -d --name mysql-master -e MYSQL_ROOT_PASSWORD=root -e MYSQL_REPLICATION_MODE=master -e MYSQL_REPLICATION_USER=repl -e MYSQL_REPLICATION_PASSWORD=root -e MYSQL_USER=my_user -e MYSQL_PASSWORD=my_password -e MYSQL_DATABASE=topscore -v \d\docker-data\mysql-master-data:/bitnami -p 3306:3306  bitnami/mysql:latest
```
**Host machine information**
```
Host Name:                 VIRENDRA
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.15063 N/A Build 15063
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Member Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Admin
Registered Organization:
Product ID:                00331-20022-15811-AA337
Original Install Date:     10/24/2017, 7:12:33 PM
System Boot Time:          3/8/2018, 10:03:33 AM
System Manufacturer:       Gigabyte Technology Co., Ltd.
System Model:              B250M-D2V
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 158 Stepping 9 GenuineIntel ~1600 Mhz
BIOS Version:              American Megatrends Inc. F5, 4/24/2017
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC+05:30) Chennai, Kolkata, Mumbai, New Delhi
Total Physical Memory:     8,080 MB
Available Physical Memory: 2,114 MB
Virtual Memory: Max Size:  9,360 MB
Virtual Memory: Available: 1,925 MB
Virtual Memory: In Use:    7,435 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    elearning.com
Logon Server:              \\SERVER
Hotfix(s):                 4 Hotfix(s) Installed.
                           [01]: KB4021572
                           [02]: KB4022405
                           [03]: KB4025376
                           [04]: KB4041676
Network Card(s):           2 NIC(s) Installed.
                           [01]: Realtek PCIe GBE Family Controller
                                 Connection Name: Ethernet
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.150.12
                                 [02]: fe80::bcb3:d7ba:1d0f:414c
                           [02]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (DockerNAT)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 10.0.75.1
                                 [02]: fe80::8040:7bd0:b0c4:46d5
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

**Describe the results you received:**
I see error so i just restart container again.

- Output of docker container:

```
Welcome to the Bitnami mysql container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mysql
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mysql/issues
Send us your feedback at containers@bitnami.com
nami    INFO  Initializing mysql
mysql   INFO  ==> Cleaning data dir...
mysql   INFO  ==> Configuring permissions...
mysql   INFO  ==> Validating inputs...
mysql   INFO  ==> Initializing database...
mysql   INFO  ==> Creating 'root' user with unrestricted access...
mysql   INFO  ==> Creating database topscore...
mysql   INFO  ==> Creating user my_user...
mysql   INFO  ==> Granting access to my_user to the database topscore...
mysql   INFO  ==> Configuring replication mode master...
mysql   INFO  ==> Creating replication user repl...
mysql   INFO  ==> Enabling remote connections...
mysql   INFO 
mysql   INFO  ########################################################################
mysql   INFO   Installation parameters for mysql:
mysql   INFO     Root User: root
mysql   INFO     Root Password: **********
mysql   INFO     Database: topscore
mysql   INFO     Username: my_user
mysql   INFO     Password: **********
mysql   INFO     Replication Mode: master
mysql   INFO     Replication Username: repl
mysql   INFO     Replication Password: **********
mysql   INFO   (Passwords are not shown for security reasons)
mysql   INFO  ########################################################################
mysql   INFO 
nami    INFO  mysql successfully initialized
INFO  ==> Starting mysql... 
nami    ERROR Unable to start com.bitnami.mysql: Cannot find pid file '/opt/bitnami/mysql/tmp/mysqld.pid'.
Welcome to the Bitnami mysql container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mysql
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mysql/issues
Send us your feedback at containers@bitnami.com
INFO  ==> Starting mysql... 
nami    ERROR Unable to start com.bitnami.mysql: Cannot find pid file '/opt/bitnami/mysql/tmp/mysqld.pid'.
```

**Version**

- Output of `docker version`:

```
Docker version 17.09.0-ce, build afdb6d4
```

- Output of `docker info`:

```
Containers: 8
 Running: 0
 Paused: 0
 Stopped: 8
Images: 25
Server Version: 17.09.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.49-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.934GiB
Name: moby
ID: OVWQ:O2PP:AZ6F:HBLK:6XDV:6KIG:C4KQ:LIGJ:ZTWB:HBYQ:HEOL:UHX3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 17
 Goroutines: 29
 System Time: 2018-03-08T07:11:20.3612706Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by juan131** [comment]
Hi @vsjadeja 

I was unable to reproduce the issue using exactly the same command, see the output below:

```
nami    INFO  Initializing mysql
mysql   INFO  ==> Cleaning data dir...
mysql   INFO  ==> Configuring permissions...
mysql   INFO  ==> Validating inputs...
mysql   INFO  ==> Initializing database...
mysql   INFO  ==> Creating 'root' user with unrestricted access...
mysql   INFO  ==> Creating database topscore...
mysql   INFO  ==> Creating user my_user...
mysql   INFO  ==> Granting access to my_user to the database topscore...
mysql   INFO  ==> Configuring replication mode master...
mysql   INFO  ==> Creating replication user repl...
mysql   INFO  ==> Enabling remote connections...
mysql   INFO
mysql   INFO  ########################################################################
mysql   INFO   Installation parameters for mysql:
mysql   INFO     Root User: root
mysql   INFO     Root Password: **********
mysql   INFO     Database: topscore
mysql   INFO     Username: my_user
mysql   INFO     Password: **********
mysql   INFO     Replication Mode: master
mysql   INFO     Replication Username: repl
mysql   INFO     Replication Password: **********
mysql   INFO   (Passwords are not shown for security reasons)
mysql   INFO  ########################################################################
mysql   INFO
nami    INFO  mysql successfully initialized
INFO  ==> Starting mysql...
2018-03-08T09:51:47.777032Z 0 [Note] Server socket created on IP: '0.0.0.0'.
2018-03-08T09:51:47.779293Z 0 [Warning] 'user' entry 'mysql.sys@localhost' ignored in --skip-name-resolve mode.
2018-03-08T09:51:47.779331Z 0 [Warning] 'db' entry 'performance_schema mysql.session@localhost' ignored in --skip-name-resolve mode.
2018-03-08T09:51:47.779343Z 0 [Warning] 'db' entry 'sys mysql.sys@localhost' ignored in --skip-name-resolve mode.
2018-03-08T09:51:47.779358Z 0 [Warning] 'proxies_priv' entry '@ root@localhost' ignored in --skip-name-resolve mode.
2018-03-08T09:51:47.780154Z 0 [Warning] 'tables_priv' entry 'user mysql.session@localhost' ignored in --skip-name-resolve mode.
2018-03-08T09:51:47.780175Z 0 [Warning] 'tables_priv' entry 'sys_config mysql.sys@localhost' ignored in --skip-name-resolve mode.
2018-03-08T09:51:47.786045Z 0 [Note] Event Scheduler: Loaded 0 events
2018-03-08T09:51:47.786384Z 0 [Note] /opt/bitnami/mysql/bin/mysqld: ready for connections.
Version: '5.7.21-log'  socket: '/opt/bitnami/mysql/tmp/mysql.sock'  port: 3306  MySQL Community Server (GPL
```

Could you please simply remove the container and retry to see if the problem still persists?

**Comment by vsjadeja** [comment]
Ok, I removed container and create new one. First time it works expected. But when I start new container after PC reboot, this will show below error:
```
INFO  ==> Starting mysql... 
nami    ERROR Unable to start com.bitnami.mysql: Cannot find pid file '/opt/bitnami/mysql/tmp/mysqld.pid'.
```
Can guide me, where I am doing wrong?

**Comment by juan131** [comment]
You're attaching a Volume to the MariaDB image (`-v \d\docker-data\mysql-master-data:/bitnami`). Is the data on this volume being persisted after you rebooted your pc?

**Comment by Ibmurai** [comment]
I'm having the same problem with "Cannot find pid file", for my slave.
Here's a snippet of the docker-compose I'm using:

```
services:
  mysql-slave:
    image: 'bitnami/mysql:5.7.21'
    ports:
      - target: 3306
        published: 3306
        mode: host
    networks:
      - network
    environment:
      - MYSQL_REPLICATION_MODE=*****
      - MYSQL_REPLICATION_USER=*****
      - MYSQL_REPLICATION_PASSWORD=*****
      - MYSQL_MASTER_HOST=*****
      - MYSQL_MASTER_PORT_NUMBER=*****
      - MYSQL_MASTER_ROOT_PASSWORD=*****
    volumes:
      - slave-storage:/bitnami/mysql/data
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure
        delay: 5s
        window: 1m
      resources:
        limits:
          cpus: '0.30'
          memory: '7.5G'
        reservations:
          cpus: '0.15'
          memory: '1G'
      placement:
        constraints: [node.role == worker]
    stop_grace_period: 5m

networks:
  network:

volumes:
  slave-storage:
```

I use the slave-storage volume, to mount from another container running xtrabackup...

**Comment by juan131** [comment]
The volume should not be a problem in your case since the PID file is not stored on the **/bitnami/mysql/data** directory where you're mounting it.

Could you please run `docker-compose up` adding the environment variable below and share the output?

```
      environment:
        - NAMI_DEBUG: "--log-level trace"
```

**Comment by Ibmurai** [comment]
It seems to happen every time the container is restarted.

There's not much to see in the logs, with trace log level :/

```
app-entrypoint.sh 08:25:05.50
app-entrypoint.sh 08:25:05.50 Welcome to the Bitnami mysql container
app-entrypoint.sh 08:25:05.50 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mysql
app-entrypoint.sh 08:25:05.50 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mysql/issues
app-entrypoint.sh 08:25:05.50 Send us your feedback at containers@bitnami.com
app-entrypoint.sh 08:25:05.50
nami    INFO  Initializing mysql
mysql   TRACE [restorePersistedData] Restoring /bitnami/mysql/data to /opt/bitnami/mysql/data
mysql   TRACE [configurePermissions] File to chown: /bitnami/mysql/data
mysql   TRACE [configurePermissions] File to chown: /opt/bitnami/mysql/tmp
mysql   TRACE [configurePermissions] File to chown: /opt/bitnami/mysql/logs
mysql   INFO
mysql   INFO  ########################################################################
mysql   INFO   Installation parameters for mysql:
mysql   INFO     Persisted data and properties have been restored.
mysql   INFO     Any input specified will not take effect.
mysql   INFO   This installation requires no credentials.
mysql   INFO  ########################################################################
mysql   INFO
nami    INFO  mysql successfully initialized
app-entrypoint.sh 08:25:15.36 INFO  ==> Starting mysql...
nami    ERROR Unable to start com.bitnami.mysql: Cannot find pid file '/opt/bitnami/mysql/tmp/mysqld.pid'.
```
...And then it exits.

**Comment by silviofdez** [comment]
Hi,

I cannot reproduce your issue. Could you please attach a script or similar in order to reproduce it?

**Comment by weimingxuan** [comment]
I also encountered this problem. The first initialization can be started, and the second initialization will be like this. Every time we need to delete the database (mysql, performance_schema, sys) and restart it.

**Comment by juan131** [comment]
Hi @weimingxuan 

Could you share with us the `docker-compose.yml` you're using and the logs you obtain when running it? It'd be also very useful if you could share what version on Bitnami MySQL image you're using.

**Comment by weimingxuan** [comment]
> Hi @weimingxuan
> 
> Could you share with us the `docker-compose.yml` you're using and the logs you obtain when running it? It'd be also very useful if you could share what version on Bitnami MySQL image you're using.

`version: '3'

services:
  web: 
    image: '793270445/php7.1-dev' 
    volumes:
      - ./web:/app
    networks: 
      - my-bridge
      
  nginx:
    image: 'bitnami/nginx:latest'
    ports:
      - '80:8080'
      - '443:8443'
    volumes:
      - ./app/nginx/vhosts:/bitnami/nginx/conf/vhosts/:ro
      - ./app/nginx/certs:/bitnami/nginx/conf/bitnami/certs
      - ./web:/app
    networks: 
      - my-bridge

  mysql:
    image: 'bitnami/mysql:latest'
    ports:
      - '3306:3306'
    environment:
      - MYSQL_ROOT_PASSWORD=root
    volumes:
      - ./data/mysql-persistence:/bitnami/mysql/data
    networks: 
      - my-bridge

networks: 
  my-bridge:
    driver: bridge`

**Comment by weimingxuan** [comment]
> Hi @weimingxuan
> 
> Could you share with us the `docker-compose.yml` you're using and the logs you obtain when running it? It'd be also very useful if you could share what version on Bitnami MySQL image you're using.

In this case, my solution is
1.docker-compose down
2.delete mysql-persistence/mysql  (/bitnami/mysql/data/mysql)
3.docker-compose up -d
It's all right to run at this time.

**Comment by juan131** [comment]
Hi @weimingxuan 

Did you another volume to persist the data after you remove **mysql-persistence** or did you disable the persistence on your solution?

**Comment by weimingxuan** [comment]
> Hi @weimingxuan
> 
> Did you another volume to persist the data after you remove **mysql-persistence** or did you disable the persistence on your solution?

I didn't do any extra work.
The computer I use is win10 system.
What are the good solutions?

**Comment by weimingxuan** [comment]
> Hi @weimingxuan
> 
> > I didn't do any extra work.
> 
> So I guess that you have data again at **"mysql-persistence"**, don't you?

I don't know why.
Before you start, you need to delete the mysql database before running properly.

**Comment by juan131** [comment]
Hi @weimingxuan 

That shouldn't be like that. You need to be able to persist the data...  Why would I use a database if I'm losing my data whenever I stop the database? 

I think the problem could be related with the mount point. You're using:

```
    volumes:
      - ./data/mysql-persistence:/bitnami/mysql/data
```

But the mount point should be **/bitnami/mysql/**:

```
    volumes:
      - ./data/mysql-persistence:/bitnami/mysql
```

**Comment by senthu-neyan** [comment]
it might be because of Mysql image version, i had slimier issue and fixed change mysql version to mysql:5.6, please, read full post.
http://www.oofitworks.com/?p=92

**Comment by juan131** [comment]
Hi @senthu-neyan 

Thanks for sharing your post.

Please note `mysql` image is not maintained by Bitnami, but Docker instead. Therefore, the solution doesn't apply to the issue we're discussing on this thread.
